include/sound/sof/topology.h

```diff
@@ -300,4 +300,9 @@ enum sof_event_types {
 	SOF_KEYWORD_DETECT_DAPM_EVENT,
 };
 
+/* extended data struct for component new */
+struct sof_ipc_comp_new_ext {
+	uint8_t uuid[16];
```

Comment:
pls add here a few reserved uint32_t fields

Reply:
I don't plan to add reserved space as traditional, 2 reasons:
1. you can't foresee how much bytes is sufficient for future use, it is a IPC bandwidth waste if it is kept 'reserved'.
2. every time you want to add a new member, just append it to the end, since we already specific the size of the extended data (hdr.size - offset), our framework can already handle the ABI un-alignment issue:
a. if driver is newer than FW, driver will send a bigger extended data struct, and FW will treat it as a smaller one (as FW extended data struct depicted) and support what extended data segments it can support only.
b. vice versa, if the FW is newer, the driver will send a smaller extended data struct, the FW will treat it as a new one but padding 0s to the end, so what we need to take care is that if the "hdr.size < extended_data_offset + item_offset + item_size", just don't trust these padding 0s.
We require the FW to have this fallback path for each new added extended data. e.g. for UUID case, we need to use component type if the SW/tplg doesn't specify component UUIDs.

Reply:
Ok, it conviced me

Reply:
Wow, I'm glad I found this explanation by chance among the 144 comments. Please add it as a comment to the source code in #2215, backward and forward compatibility are tricky and need at least that.

See related hdr.size suggestion in https://github.com/thesofproject/linux/pull/2215#issuecomment-648584786